include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3260,6 +3296,43 @@ namespace internal
 
 
 
+  template <int dim, int length, typename Number2, typename Number>
+  inline DEAL_II_ALWAYS_INLINE void
+  do_test_xy(AlignedVector<Number2> &                    values,
```

Comment:
I like the fact that you do not use `integrate`, as this is really just the addition of a quantity multiplied by all test functions, accumulated into a field. But I think `test` alone is not expressive enough, so maybe
```suggestion
  do_apply_test_functions_xy(AlignedVector<Number2> &                    values,
```